iceoryx_examples/iceperf/iceoryx.cpp

```diff
@@ -68,24 +69,29 @@ void Iceoryx::shutdown() noexcept
 
 void Iceoryx::sendPerfTopic(uint32_t payloadSizeInBytes, bool runFlag) noexcept
 {
-    auto sendSample = static_cast<PerfTopic*>(m_publisher.allocateChunk(payloadSizeInBytes, true));
-    sendSample->payloadSize = payloadSizeInBytes;
-    sendSample->run = runFlag;
-    sendSample->subPackets = 1;
-
-    m_publisher.sendChunk(sendSample);
+    m_publisher.loan(payloadSizeInBytes).and_then([&](iox::popo::Sample<void>& sample) {
+        auto sendSample = static_cast<PerfTopic*>(sample.get());
+        sendSample->payloadSize = payloadSizeInBytes;
+        sendSample->run = runFlag;
+        sendSample->subPackets = 1;
+        sample.publish();
+    });
 }
 
 PerfTopic Iceoryx::receivePerfTopic() noexcept
 {
     const void* receivedChunk;
-    while (!m_subscriber.getChunk(&receivedChunk))
+    PerfTopic receivedSample;
+    while (!m_subscriber.take().and_then([&](iox::cxx::optional<iox::popo::Sample<const void>>& maybeSample) {
```

Comment:
I think this is an error. The the `!` operator is applied to the `operator bool` from the expected, not the optional. At least currently. Might change after #352 and the semantic will change after #351

Reply:
Thanks for the heads-up. This line is pretty convoluted :/

Reply:
Now using a designated `bool` for checking and not the `bool operator.